crates/backend/src/codegen.rs

```diff
@@ -1571,13 +1576,13 @@ impl ToTokens for ast::Enum {
             #[automatically_derived]
             impl #wasm_bindgen::convert::OptionFromWasmAbi for #enum_name {
                 #[inline]
-                fn is_none(val: &u32) -> bool { *val == #hole }
+                fn is_none(val: &Self::Abi) -> bool { *val == #hole as #underlying }
```

Comment:
The `as` conversion here shouldn't be necessary, right?

Reply:
No, it's necessary because `Self::Abi` is either `u32` or `i32` while `hole` is always `u32`. So this would be a type error without the cast. This is because `quote` preserves the type of interpolated values. I.e. [this is how tokens are generated for u32](https://docs.rs/quote/latest/src/quote/to_tokens.rs.html#176-180).

Reply:
I see, thanks!